pkg/lib/serverTime.js

```diff
@@ -258,41 +268,34 @@
         return status;
     };
 
-    const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
-                                                { superuser: "try" });
-
-    self.get_custom_ntp = function () {
-        /* We only support editing the configuration of
-         * systemd-timesyncd, by dropping a file into
-         * /etc/systemd/timesyncd.conf.d.  We assume that timesyncd is
-         * used when:
-         *
-         * - systemd-timedated is answering for
-         *   org.freedesktop.timedate1 as opposed to, say, timedatex.
-         *
-         * - systemd-timesyncd is enabled (false if chrony is being used)
-         *
-         * The better alternative would be to have an API in
-         * o.fd.timedate1 for managing the list of NTP server
-         * candidates.
-         */
-        const result = {
-            supported: false,
-            enabled: false,
-            servers: []
-        };
+    function service_available(s) {
+        return s.exists && s.unit.LoadState != "masked";
+    }
 
-        if (!timedate1_service.exists || timedate1_service.unit.Id !== "systemd-timedated.service") {
-            console.log("systemd-timedated not in use, ntp server configuration not supported");
-            return Promise.resolve(result);
+    function get_timesync_backend() {
+        // We assume chronyd > ntpd > timesyncd.  If the assumption
+        // turns out to be wrong, we will tell the user afterwards.
+        if (service_available(chronyd_service)) {
+            return "chronyd";
+        } else if (service_available(ntpd_service)) {
+            return null;
+        } else if (service_available(timesyncd_service)) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230123-110321-058761ba-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)